app/assets/javascripts/components/common/confirm.jsx

```diff
@@ -1,29 +1,63 @@
 import React from 'react';
 import Modal from './modal.jsx';
+import TextInput from './text_input.jsx';
 
 const Confirm = React.createClass({
   displayName: 'Confirm',
 
   propTypes: {
     onConfirm: React.PropTypes.func.isRequired,
     onCancel: React.PropTypes.func.isRequired,
-    message: React.PropTypes.string.isRequired
+    message: React.PropTypes.string.isRequired,
+    showInput: React.PropTypes.bool,
+    passcode: React.PropTypes.string,
```

Comment:
We should name all the props in a general way, so that they don't refer specifically to this use case. In this case, I don't think we need to have a prop for this at all. We can just pass the state to the onConfirm function.